ci/scripts/python_test.sh

```diff
@@ -30,4 +30,14 @@ export ARROW_GDB_SCRIPT=${arrow_dir}/cpp/gdb_arrow.py
 # Enable some checks inside Python itself
 export PYTHONDEVMODE=1
 
+# By default, force-test all optional components
+export PYARROW_TEST_CUDA=${PYARROW_TEST_ORC:-${ARROW_CUDA}}
+export PYARROW_TEST_DATASET=${PYARROW_TEST_ORC:-${ARROW_DATASET}}
+export PYARROW_TEST_FLIGHT=${PYARROW_TEST_ORC:-${ARROW_FLIGHT}}
+export PYARROW_TEST_GANDIVA=${PYARROW_TEST_ORC:-${ARROW_GANDIVA}}
+export PYARROW_TEST_HDFS=${PYARROW_TEST_ORC:-${ARROW_HDFS}}
+export PYARROW_TEST_ORC=${PYARROW_TEST_ORC:-${ARROW_ORC}}
+export PYARROW_TEST_PARQUET=${PYARROW_TEST_ORC:-${ARROW_PARQUET}}
+export PYARROW_TEST_S3=${PYARROW_TEST_ORC:-${ARROW_S3}}
```

Comment:
Shouldn't these look like `export PYARROW_TEST_S3=${ARROW_S3:-ON}`?

Reply:
No, we want to be able to override testing, I think? However, it seems I got the above wrong :-)

Reply:
Ohh, right: `: ${PYARROW_TEST_S3:=${ARROW_S3:-ON}}` ?

Reply:
Oh, yes, exactly.